tools/integration_tests/util/setup/setup.go

```diff
@@ -642,3 +647,38 @@ func CreateProxyServerLogFile(t *testing.T) string {
 func AppendProxyEndpointToFlagSet(flagSet *[]string, port int) {
 	*flagSet = append(*flagSet, "--custom-endpoint="+fmt.Sprintf("http://localhost:%d/storage/v1/", port))
 }
+
+// GetGCEZone returns the GCE zone of the current machine from
+// the GCP resource detector.
+func GetGCEZone(ctx context.Context) (string, error) {
+	detectedAttrs, err := resource.New(ctx, resource.WithDetectors(gcp.NewDetector()))
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch GCP resource detector: %w", err)
+	}
+	attrs := detectedAttrs.Set()
+	if zoneValue, exists := attrs.Value("cloud.availability_zone"); exists {
+		zone := zoneValue.AsString()
+		// Confirm that the zone string is in right format e.g. us-central1-a .
+		if match, err := regexp.MatchString(zoneMatcherRegex, zone); !match || err != nil {
+			return zone, fmt.Errorf("zone %q returned by GCP resource detector is not a valid zone-string: %w", zone, err)
+		}
+		return zone, nil
+	}
+	return "", fmt.Errorf("cloud.availability_zone not found in GCP resource detector")
+}
+
+// GetGCERegion return the GCE region for a given GCE zone.
+// E.g. from us-central1-a, it returns us-central1 .
+func GetGCERegion(gceZone string) (string, error) {
+	indexOfLastHyphen := strings.LastIndex(gceZone, "-")
+	if indexOfLastHyphen < 0 {
+		return "", fmt.Errorf("Input gceZone %q is not proper. It is expected to be of the form <country>-<region>-<zone> e.g. us-central1-a .", gceZone)
```

Comment:
```suggestion
		return "", fmt.Errorf("input gceZone %q is not proper. It is expected to be of the form <country>-<region>-<zone> e.g. us-central1-a .", gceZone)
```
Error string shouldn't be capitalized: go/go-style/decisions#error-strings